src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -362,7 +362,7 @@ public function symlink(string $originDir, string $targetDir, bool $copyOnWindow
      * @throws FileNotFoundException When original file is missing or not a file
      * @throws IOException           When link fails, including if link already exists
      */
-    public function hardlink(string $originFile, $targetFiles)
+    public function hardlink(string $originFile, string|array $targetFiles)
```

Comment:
```suggestion
    public function hardlink(string $originFile, string|iterable $targetFiles)
```

(consistent with other functions in this class that use `toIterable()`.